qualtran/bloqs/factoring/ecc/ec_add_r.py

```diff
@@ -67,6 +85,41 @@ def signature(self) -> 'Signature':
             [Register('ctrl', QBit()), Register('x', QUInt(self.n)), Register('y', QUInt(self.n))]
         )
 
+    def build_composite_bloq(
```

Comment:
where does this decomposition come from? it does not match the references.

Reply:
I'm a little confused myself on whether ECAddR and ECWindowAddR are necessary distinctions. Is ECAddR not just supposed to be a single window ECWindowAddR? In that case we can make it one bloq with a var window_size that changes decomposition. Otherwise I couldn't find a special decomposition for the ECAddR circuit so I put what I imagined it would be (initialize a,b from R.x,R.y; ECPointAdd; uninitialize a,b;).

Reply:
I think we chatted in person, but the Roetteler / q# reference has a different circuit for this. What do you want to do here? I think we discussed having the "main call graph" go via ECWindowAddR. We'll keep the reference to the special circuit for adding a constant (without additional register allocation) but not use it